src/indicators/drought/ndvi.py

```diff
@@ -0,0 +1,192 @@
+"""Code to download and process NDVI data
+
+NDVI data is downloadable and available through the USGS
+with a description available at
+https://earlywarning.usgs.gov/fews/product/451 and the
+actual data from the USGS file explorer:
+https://edcintl.cr.usgs.gov/downloads/sciweb1/shared/fews/web/africa/west/dekadal/emodis/ndvi_c6/temporallysmoothedndvi/downloads/monthly/
+
+The products include temporally smoothed NDVI, median anomaly,
+difference from the previous year, and median anomaly
+presented as a percentile. For the current exploration,
+will just use percent of median because it's similar
+to the % anomaly products we are comparing to for Biomasse
+and WRSI in Chad. However, can expand this code more easily in
+the future to download and process other NDVI data.
+
+Data is published with about 1 month delay from the end of the
+dekad.
+"""
+import logging
+import os
+from datetime import date, datetime
+from io import BytesIO
+from pathlib import Path
+from typing import List, Union
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import pandas as pd
+import rioxarray  # noqa: F401
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+_BASE_URL = (
+    "https://edcintl.cr.usgs.gov/downloads/sciweb1/shared/fews/"
+    "web/africa/west/dekadal/emodis/ndvi_c6/percentofmedian/downloads/dekadal/"
+)
+
+_BASE_FILENAME = "wa{year:02}{dekad:02}pct"
+
+_RAW_DIR = Path(os.getenv("AA_DATA_DIR"), "public", "raw", "glb", "ndvi")
+
+
+def _get_paths(year, dekad):
+    base_file_name = _BASE_FILENAME.format(year=year % 100, dekad=dekad)
+    url_zip_filename = base_file_name + ".zip"
+    url_path = _BASE_URL + url_zip_filename
+    file_name = base_file_name + ".tif"
+    return url_path, file_name
+
+
+def _download_ndvi(year: int, dekad: int):
+    url, file_name = _get_paths(year, dekad)
+    try:
+        resp = urlopen(url)
+    except HTTPError:
+        logger.error(
+            f"No NDVI data available for "
+            f"dekad {dekad} of {year}, skipping."
+        )
+        return
+
+    zf = ZipFile(BytesIO(resp.read()))
+    save_path = Path(_RAW_DIR, file_name)
+    save_path.unlink(missing_ok=True)
+
+    zf.extract(file_name, _RAW_DIR)
+
+
+def _date_to_dekad(date_obj: Union[date, str]):
+    if isinstance(date_obj, str):
+        date_obj = date.fromisoformat(date_obj)
+    year = date_obj.year
+    dekad = (
+        (date_obj.timetuple().tm_mday // 10) + ((date_obj.month - 1) * 3) + 1
```

Comment:
Maybe I am misinterpreting but why would you use `date_obj.timetuple().tm_mday` instead of `date_obj.day` ? Also applies to the same function in `wrsi.py`